### PR TITLE
#96 - mongodb fooditem category

### PR DESCRIPTION
**Pull request by flashback2k14**
implement Base Repository to reduce duplicated code

### CODE REVIEW

File: server/src/food/handler/FoodItemCategory.handler.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      const result = await this._db.fooditemCategory.getAll();
+      res.status(200).json(result);
+    } catch (error) {
+      res.status(error.statusCode).json(error.forResponse());
+    }
+  }
+
+  public getById = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const id = new Types.ObjectId(req.params.id);
```
Comment (hetec) [inline server/src/food/handler/FoodItemCategory.handler.ts]: It's not necessary to create a ObjectId Instance your can simply use the string

File: server/src/food/handler/FoodItemCategory.handler.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    } catch (error) {
+      res.status(error.statusCode).json(error.forResponse());
+    }
+  }
+
+  public save = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const category = new FoodItemCategory().clone(req.body);
+
+      // TODO:  await this._db.fooditem.getById(category.foodItemId);
```
Comment (hetec) [inline server/src/food/handler/FoodItemCategory.handler.ts]: What's about this todo?

File: server/src/food/handler/FoodItemCategory.handler.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }
+
+  public updateById = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const categoryId = new Types.ObjectId(req.params.id);
+      const foodItemId = req.body.foodItemId;
+      const updatedCategoryName = req.body.name;
+
+      // TODO:  await this._db.fooditem.getById(foodItemId);
```
Comment (hetec) [inline server/src/food/handler/FoodItemCategory.handler.ts]: same as above

File: server/src/food/routes/FoodItemCategory.route.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    this._router.get("/",
+      this._authValidationHandler.checkToken,
+      this._fooditemCategoryHandler.getAll);
+
+    this._router.get("/:id",
+      this._authValidationHandler.checkToken,
+      this._fooditemCategoryHandler.getById);
+
+    this._router.put("/:id",
+      this._authValidationHandler.checkToken,
```
Comment (hetec) [inline server/src/food/routes/FoodItemCategory.route.ts]: Is everybody allowed to create, update and delete categories? Maybe it's better to provide the users with a consistent predefined set of categories and only allow admins to update this list. What do you think?

File: server/src/food/handler/FoodItemCategory.handler.ts (hunk 1 above)
Comment (flashback2k14) [reply server/src/food/handler/FoodItemCategory.handler.ts]: yes, i know but i have always used ObjectId as type.

File: server/src/food/handler/FoodItemCategory.handler.ts (hunk 2 above)
Comment (flashback2k14) [reply server/src/food/handler/FoodItemCategory.handler.ts]: it's not implemented at this time. that's the next step.

File: server/src/food/handler/FoodItemCategory.handler.ts (hunk 3 above)
Comment (flashback2k14) [reply server/src/food/handler/FoodItemCategory.handler.ts]: it's not implemented at this time. that's the next step.

File: server/src/food/routes/FoodItemCategory.route.ts (hunk 4 above)
Comment (flashback2k14) [reply server/src/food/routes/FoodItemCategory.route.ts]: that's a good idea. i will add this check.

Comment (flashback2k14) [reply server/src/food/routes/FoodItemCategory.route.ts]: what is with fooditem unit and fooditem template? should we restrict this too?

Comment (hetec) [reply server/src/food/routes/FoodItemCategory.route.ts]: I'm not sure but at the moment I would say no. Let's keep this in mind

**Pull request merged**